crates/services/sync/src/import.rs

```diff
@@ -364,28 +399,72 @@ fn get_block_stream<
     params: Config,
     p2p: Arc<P>,
     consensus: Arc<C>,
+    cache: SharedMutex<HashMap<Range<u32>, CachedData>>,
 ) -> impl Stream<Item = impl Future<Output = SealedBlockBatch>> {
-    let header_stream = get_header_batch_stream(range.clone(), params, p2p.clone());
-    header_stream
+    let ranges = range_chunks(range, params.header_batch_size);
```

Comment:
Stylistic nit, because this function is now too much of a blob for me.

Please consider transforming it into something like the example below, as long as cloning all the arcs and mutexes isn't too much of a hassle.

```
futures::stream::iter(ranges)
        .map({
            move |range| {
                fetch_headers(Arc::clone(&p2p), SharedMutex::clone(&cache), range)
            }
        })
        .map({
            move |header_batch| {
                check_headers(
                    Arc::clone(&p2p),
                    Arc::clone(&consensus),
                    SharedMutex::clone(&cache),
                    header_batch,
                )
            }
        })
        .map({
            move |headers| {
                fetch_blocks(headers, consensus, p2p, cache)
                    .instrument(tracing::debug_span!("consensus_and_transactions"))
                    .in_current_span()
            }
        })
```

Reply:
I agree and there is a lot more things to improve on this service I don't wanna make this PR even bigger and so I created an issue for that : https://github.com/FuelLabs/fuel-core/issues/2370